components/script/dom/webglframebuffer.rs

```diff
@@ -289,12 +290,37 @@ impl WebGLFramebuffer {
             };
 
             if matched {
-                *attachment.borrow_mut() = None;
-                self.update_status();
+                closure(attachment);
             }
         }
     }
 
+    pub fn detach_renderbuffer(&self, rb: &WebGLRenderbuffer) {
+        self.call_on_matching_renderbuffers(rb, |att| {
+            *att.borrow_mut() = None;
+            self.update_status();
```

Comment:
Neat idea.  I think I'll stick with the simple version here, since you couldn't have a complete FBO with the same object attached twice anyway.